libp2p/nameresolving/nameresolver.nim

```diff
@@ -118,7 +118,7 @@ proc resolveMAddress*(
   if not DNS.matchPartial(address):
     res.incl(address)
   else:
-    let code = address[0].get().protoCode().get()
+    let code = address[0].tryGet().protoCode().tryGet()
```

Comment:
do we want to raise a `Defect` at line 132?

Reply:
Yes, we just checked above that this wasn't possible (with the matchPartial)

Reply:
But there could be a bug in the code, don't we wanna get rid of Defects?